### PR TITLE
MM-29969: add e2e test to pluginAddCmdF command

### PR DESCRIPTION
**Pull request by jecepeda**
#### Summary

This PR adds tests to pluginAddCmdF command.
Cases covered:
 - Config doesn't allow plugin upload
 - Config allowed: admin and local user can upload plugin. Normal user can't

#### Ticket Link

- https://mattermost.atlassian.net/browse/MM-29969
- Fixes https://github.com/mattermost/mattermost-server/issues/16074

### Concerns

- In order to get the example plugin I had to use the environment variable `MM_SERVER_PATH`. Is that the correct way? Using `./path_to_the_file` didn't work for me.

### CODE REVIEW

File: commands/plugin_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+func (s *MmctlE2ETestSuite) TestPluginAddCmd() {
+	s.SetupTestHelper().InitBasic()
+
+	mmPath := os.Getenv("MM_SERVER_PATH")
+	pluginPath := filepath.Join(mmPath, "../mmctl/commands/test_files/testplugin.tar.gz")
+	fmt.Println(pluginPath)
+
+	s.RunForSystemAdminAndLocal("admin and local can't add plugins if the config doesn't allow it", func(c client.Client) {
+		printer.Clean()
```
Comment (isacikgoz) [inline commands/plugin_e2e_test.go:28]: Nit: We can have an empty line after this one, for the consistency.

File: commands/plugin_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+		// teardown
+		pInfo := res.Inactive[0]
+		appErr = s.th.App.RemovePlugin(pInfo.Id)
+		s.Require().Nil(appErr)
+
+		s.th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.PluginSettings.Enable = false
+			*cfg.PluginSettings.EnableUploads = false
+		})
```
Comment (isacikgoz) [inline commands/plugin_e2e_test.go:60]: We can do this with a `defer`.

File: commands/plugin_e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		err := pluginAddCmdF(s.th.Client, &cobra.Command{}, []string{pluginPath})
+		s.Require().Nil(err)
+		s.Require().Equal(1, len(printer.GetErrorLines()))
+		s.Require().Contains(printer.GetErrorLines()[0], "You do not have the appropriate permissions")
+
+		// teardown
+		s.th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.PluginSettings.Enable = false
+			*cfg.PluginSettings.EnableUploads = false
+		})
```
Comment (isacikgoz) [inline commands/plugin_e2e_test.go:79]: Same as above.

**Review by isacikgoz — APPROVED** [review]
LGTM, Have some minor comments. Thanks for the contribution @jecepeda 🚀🎉

File: commands/plugin_e2e_test.go [hunk 4]
```diff
@@ -5,14 +5,84 @@ package commands
...
 	"github.com/mattermost/mmctl/client"
 	"github.com/mattermost/mmctl/printer"
 )
 
+func (s *MmctlE2ETestSuite) TestPluginAddCmd() {
+	s.SetupTestHelper().InitBasic()
+
+	mmPath := os.Getenv("MM_SERVER_PATH")
+	pluginPath := filepath.Join(mmPath, "../mmctl/commands/test_files/testplugin.tar.gz")
+	fmt.Println(pluginPath)
```
Comment (mgdelacroix) [inline commands/plugin_e2e_test.go:25]: We can directly use the `testplugin` file that exists in the server repository and not add it to `mmctl`:

```suggestion
	pluginPath := filepath.Join(os.Getenv("MM_SERVER_PATH"), "tests", "testplugin.tar.gz")
```

File: commands/plugin_e2e_test.go [hunk 5]
```diff
@@ -5,14 +5,84 @@ package commands
...
+
+		res, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Equal(1, len(res.Inactive))
+
+		// teardown
+		pInfo := res.Inactive[0]
+		appErr = s.th.App.RemovePlugin(pInfo.Id)
+		s.Require().Nil(appErr)
+
```
Comment (mgdelacroix) [inline commands/plugin_e2e_test.go:63]: nit: unnecessary empty line

**Review by mgdelacroix — CHANGES_REQUESTED** [review]
Thanks for the contribution @jecepeda!! It's looking really good, added only one change request not to have the plugin file in the repo, taking advantage of the one we have in the `mattermost-sever` repository

**Review by mgdelacroix — APPROVED** [review]
LGTM! Thanks for applying the changes @jecepeda!

**Review by isacikgoz — APPROVED** [review]
Awesome!

**Review by lindy65 — APPROVED** [review]
Thanks @jecepeda @isacikgoz - looks good to me 👍

**Pull request merged**